### PR TITLE
AOA and SSA receive, AOA indication, bulls eye indication

### PR DESCRIPTION
**Pull request by EShamaev**
Preliminary PR for AOA and SSA estimation receive and display testing.

Depends on merging of:
https://github.com/ArduPilot/mavlink/pull/32
https://github.com/ArduPilot/ardupilot/pull/5933

### CODE REVIEW

**Comment by EShamaev** [comment]
https://cloud.githubusercontent.com/assets/7996972/24836783/eba4dbb0-1d2d-11e7-8630-8d4f37211320.png

File: ExtLibs/Mavlink/message_definitions/common.xml [hunk 1]
```diff
@@ -3587,6 +3587,12 @@
       <field type="float" name="size_y">Size in radians of target along y-axis</field>
     </message>
     <!-- MESSAGE IDs 180 - 229: Space for custom messages in individual projectname_messages.xml files -->
+    <message id="227" name="AOA_SSA">
```
Comment (magicrub) [inline ExtLibs/Mavlink/message_definitions/common.xml]: range 180-229 is reserved.

**Comment by EShamaev** [comment]
Moved to higher address space. Test will be done after installation of MSVS 15 as in my old 14 MP does not compile after change to .NET 4.6.2.

**Comment by EShamaev** [comment]
Rebased and squashed.
As the message is in MavLink V2, this PR is also needed: https://github.com/ArduPilot/MissionPlanner/pull/1532

**Comment by EShamaev** [comment]
Please advice if I should rework commits to be one module(file?)/one commit or the way it is now is fine?

File: ParameterMetaDataBackup.xml [hunk 2]
```diff
@@ -19721,5 +19721,11 @@
       <Range>1 255</Range>
       <User>Advanced</User>
     </CMD_TOTAL>
+    <AOA_CRIT>
```
Comment (meee1) [inline ParameterMetaDataBackup.xml]: you have places this under ardutracker, there are multiple vechile class's wityhin this file.
normally I generate this off latest stable also. so can be skipped in the commit

File: ParameterFactMetaData.xml [hunk 3]
```diff
@@ -4941,5 +4941,11 @@ Maps the change of airspeed error to the acceleration setpoint</short_desc>
       <min>0</min>
       <max>18</max>
     </parameter>
+    <parameter default="25" name="AOA_CRIT" type="FLOAT">
```
Comment (meee1) [inline ParameterFactMetaData.xml]: this can be skipped as well as its for px4

File: ExtLibs/Mavlink/message_definitions/ardupilotmega.xml [hunk 4]
```diff
@@ -1390,5 +1390,12 @@
       <field name="AOA" type="float">Angle of Attack (degrees)</field>
       <field name="SSA" type="float">Side Slip Angle (degrees)</field>
     </message>
+    <!-- Angle of Attack and Side Slip Angle message -->
```
Comment (meee1) [inline ExtLibs/Mavlink/message_definitions/ardupilotmega.xml]: I would normally grab this from upstream this ensures whatever is in upstream mp supports with the approx. checking time

**Comment by EShamaev** [comment]
Adjusted files as per comments.
ardupilotmega.xml needs to be taken from upstream and regenerated as it is not included in this PR.

**Comment by EShamaev** [comment]
Rebased to latest master

**Comment by EShamaev** [comment]
Thank you!!!

**Pull request merged**